### DISCUSSION TITLE
How to read embedded json data

### DISCUSSION THREAD

**Discussion by milasudril** [nlohmann/json#3720] [Q&A]
I want to read embedded json data, like this:

```
some json
other data
more json
more other data
```
I tried this:
```c++
#include <nlohmann/json.hpp>

using json = nlohmann::json;

int main()
{
	auto data = json::parse(stdin);
}
```

```
$  ./a.out
{"test":"foo"}more data 
terminate called after throwing an instance of 'nlohmann::detail::parse_error'
  what():  [json.exception.parse_error.101] parse error at line 2, column 1: syntax error while parsing value - invalid literal; last read: '"foo"}<U+000A>m'; expected end of input
  ```

After successfully loaded the first object, the next character read from stdin should be the newline, and data should be equal to {"test":"foo"}. That is, the parser should stop when all object levels have been closed.

I know that this is not according to the JSON specification, but it is a *very* useful feature.

**Comment by nlohmann** [comment]
See https://json.nlohmann.me/features/parsing/json_lines/

**Reply by milasudril** [reply]
What about something like `JSON_DISABLE_EOF_CHECK` `JSON_DECODE_ANY`?

**Reply by falbrechtskirchinger** [reply]
That's already possible with the SAX API. We should make it easier, but that would likely require a breaking change and will have to wait till 4.0.0.
